Tests/ClientTests/Autofill/CreditCard/ViewComponents/CreditCardInputViewModelTests.swift

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import XCTest
+@testable import Client
+
+class CreditCardInputViewModelTests: XCTestCase {
+    override class func setUp() {
+        super.setUp()
+    }
+
+    override class func tearDown() {
+        super.tearDown()
+    }
+
+    func testIsNameValid() {
+        let viewModel = CreditCardInputViewModel(fieldType: .name)
+
+        viewModel.updateNameValidity(userInputtedText: "Tim Apple")
+        XCTAssert(viewModel.isValid)
+
+        viewModel.updateNameValidity(userInputtedText: "")
+        XCTAssertFalse(viewModel.isValid)
+
+        // These tests will be uncommented when these cases are addressed.
+        // Currently, we have only basic checks but will update as the tasks are
+        // created.
+//        viewModel.updateNameValidity(userInputtedText: "1022992")
+//        XCTAssertFalse(viewModel.isValid)
+//
+//        viewModel.updateNameValidity(userInputtedText: ")(*&^%$#@")
+//        XCTAssertFalse(viewModel.isValid)
+    }
+
+    func testIsCardNumberValid() {
+        let viewModel = CreditCardInputViewModel(fieldType: .number)
+
+        viewModel.updateCardNumberValidity(userInputtedText: "1234123412341234")
+        XCTAssert(viewModel.isValid)
+
+        viewModel.updateCardNumberValidity(userInputtedText: "123412341234123")
+        XCTAssert(viewModel.isValid)
+
+        viewModel.updateCardNumberValidity(userInputtedText: "")
+        XCTAssertFalse(viewModel.isValid)
+    }
+
+    func testIsExpirationValid() {
```

Comment:
Same comment as above break down into separate test func with significantly names indicating what is being tested this way it will be easier to read and understand the test